webpack/assets/javascripts/react_app/components/HostsIndex/index.js

```diff
@@ -0,0 +1,155 @@
+  KebabToggle,
+  DropdownItem,
+  ToolbarItem,
+} from '@patternfly/react-core';
+import { translate as __ } from '../../common/I18n';
+import TableIndexPage from '../PF4/TableIndexPage/TableIndexPage';
+import { ActionKebab } from '../PF4/TableIndexPage/ActionKebab';
+import { HOSTS_API_PATH, API_REQUEST_KEY } from '../../routes/Hosts/constants';
+import { useAPI } from '../../common/hooks/API/APIHooks';
+import { useBulkSelect } from '../PF4/TableIndexPage/Table/TableHooks';
+import SelectAllCheckbox from '../PF4/TableIndexPage/Table/SelectAllCheckbox';
+import { getPageStats } from '../PF4/TableIndexPage/Table/helpers';
+import { STATUS } from '../../constants';
+
+const HostsIndex = () => {
+  const columns = {
+    name: {
+      title: __('Name'),
+      wrapper: ({ id, name }) => <a href={`hosts/${id}`}>{name}</a>,
+      isSorted: true,
+    },
+  };
+  const defaultParams = { search: '' }; // search ||
+
+  const response = useAPI(
+    'get',
+    HOSTS_API_PATH.includes('include_permissions')
+      ? HOSTS_API_PATH
+      : `${HOSTS_API_PATH}?include_permissions=true`,
+    {
+      ...{ key: API_REQUEST_KEY },
+      params: defaultParams,
+    }
+  );
+
+  const {
+    response: {
+      search: apiSearchQuery,
+      can_create: canCreate,
+      results,
+      total,
+      per_page: perPage,
+      page,
+      subtotal,
+      message: errorMessage,
+    },
+    status = STATUS.PENDING,
+    setAPIOptions,
+  } = response;
+
+  const { pageRowCount } = getPageStats({ total, page, perPage });
+
+  const {
+    updateSearchQuery,
+    fetchBulkParams,
+    ...selectAllOptions
+  } = useBulkSelect({
+    results,
+    metadata: {},
+  });
+
+  const {
+    selectAll,
+    selectPage,
+    selectNone,
+    selectedCount,
+    selectOne,
+    areAllRowsOnPageSelected,
+    areAllRowsSelected,
+    isSelected,
+  } = selectAllOptions;
+
+  const computeContentSource = search =>
+    `/change_host_content_source?search=${search}`;
+
+  const item = {
+    title: __('Change content source'),
+    icon: <CubeIcon />,
+    computeHref: computeContentSource,
+  };
+
+  const selectionToolbar = (
+    <ToolbarItem key="selectAll">
+      <SelectAllCheckbox
+        {...{
+          selectAll,
+          selectPage,
+          selectNone,
+          selectedCount,
+          pageRowCount,
+        }}
+        totalCount={total}
+        areAllRowsOnPageSelected={areAllRowsOnPageSelected()}
+        areAllRowsSelected={areAllRowsSelected()}
+      />
+    </ToolbarItem>
+  );
+
+  const rowSelectNodeFunc = result => (
+    <Td
+      select={{
+        rowIndex: result.id,
+        onSelect: (_event, isSelecting) => {
+          selectOne(isSelecting, result.id);
+        },
+        isSelected: isSelected(result.id),
+        disable: false,
+      }}
+    />
+  );
+
+  const computeHref = () => {
+    if (selectedCount > 0) {
+      return `/change_host_content_source?search=${fetchBulkParams()}`;
+    }
+    return '';
+  };
```

Comment:
This doesn't need to be a function

```suggestion
  const dropdownHref = selectedCount > 0 ? `/change_host_content_source?search=${fetchBulkParams()}` : '';
```